bears/js/JSHintBear.py

```diff
@@ -16,19 +31,377 @@ class JSHintBear(LocalBear, Lint):
+                                                 - false   - same as inner
+                                                 - true    - allow variable
+                                                             shadowing
+        :param prohibit_unused:                 This option genrates warnigns
+                                                when variables are defined but
+                                                never used. This can be set to
+                                                ``vars`` to only check for
+                                                variables, not function
+                                                parameters,or ``strict`` to
+                                                check all variables and
+                                                parameters.
+        :param allow_latedef:                   This option prohibits the use of
+                                                a variable before it was defined
+                                                Setting this option to "nofunc"
+                                                will allow function declarations
+                                                to be ignored.
+        :param es_version:                      This option is used to specify
+                                                the ECMAScript version to which
+                                                the code must adhere to.
+        :param jshint_config:                   The location of the jshintrc
+                                                config file. If this option is
+                                                present all the above options
+                                                are not used. Instead the
+                                                .jshintrc file is used as the
+                                                configuration file.
         '''
         self.arguments = '--verbose {filename}'
+
         if jshint_config:
+            self.config_options = None
             self.arguments += (" --config "
                                + escape_path_argument(jshint_config))
+        else:
+            options = {
+                "bitwise": prohibit_bitwise,
+                "freeze": prohibit_prototype_overwrite,
+                "curly": force_braces,
+                "eqeqeq": prohibit_type_coercion,
+                "futurehostile": future_hostile,
+                "notypeof": prohibit_typeof,
+                "es3": es3,
+                "forin": force_filter_forin,
+                "funcscope": allow_funcscope,
+                "iterator": iterator,
+                "noarg": prohibit_arg,
+                "nocomma": prohibit_comma,
+                "nonbsp": prohibit_non_breaking_whitespace,
+                "nonew": prohibit_new,
+                "undef": prohibit_undefined,
+                "singleGroups": prohibit_groups,
```

Comment:
is this one correct? all other keys are completely lowercase^^ so not `singlegroups`?